sstables/sstable_set.cc

```diff
@@ -177,6 +177,29 @@ sstable_set::insert(shared_sstable sst) {
     _impl->insert(sst);
 }
 
+// Default implementation for sstable_set_impl vector insert
+void
+sstable_set_impl::insert(const std::vector<shared_sstable>& ssts) {
+    std::vector<shared_sstable> undo_list;
+    undo_list.reserve(ssts.size());
+    try {
+        for (const auto& sst : ssts) {
+            insert(sst);
+            undo_list.emplace_back(sst);
```

Comment:
The undo vector is not needed, just the number of sstables successfully inserted.